src/main/java/org/opensearch/knn/quantization/quantizer/MultiBitScalarQuantizer.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.knn.quantization.quantizer;
+
+import org.opensearch.knn.quantization.models.quantizationOutput.BinaryQuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationParams.SQParams;
+import org.opensearch.knn.quantization.models.quantizationState.MultiBitScalarQuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+import org.opensearch.knn.quantization.models.requests.TrainingRequest;
+import org.opensearch.knn.quantization.sampler.Sampler;
+import org.opensearch.knn.quantization.sampler.SamplingFactory;
+import org.opensearch.knn.quantization.util.BitPacker;
+import org.opensearch.knn.quantization.util.QuantizerHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * MultiBitScalarQuantizer is responsible for quantizing vectors into multi-bit representations per dimension.
+ * It supports multiple bits per coordinate, allowing for finer granularity in quantization.
+ */
+public class MultiBitScalarQuantizer implements Quantizer<float[], byte[]> {
+    private final int bitsPerCoordinate; // Number of bits used to quantize each dimension
+    private final int samplingSize; // Sampling size for training
+    private final Sampler sampler; // Sampler for training
+    private static final boolean IS_TRAINING_REQUIRED = true;
+    // Currently Lucene has sampling size as
+    // 25000 for segment level training , Keeping same
+    // to having consistent, Will revisit
+    // if this requires change
+    private static final int DEFAULT_SAMPLE_SIZE = 25000;
+
+    /**
+     * Constructs a MultiBitScalarQuantizer with a specified number of bits per coordinate.
+     *
+     * @param bitsPerCoordinate the number of bits used per coordinate for quantization.
+     */
+    public MultiBitScalarQuantizer(final int bitsPerCoordinate) {
```

Comment:
Do we need multiple constructors or can we just have one?

Reply:
I had this for easy configuration till we have multiple version for Sampler , and I wanted to avoid passing sampler from Registry , It does not harm with Overloaded Constructor.